### ISSUE TITLE
DR-3 Restart Miner Functionality

### ISSUE THREAD

**Issue by hibes** [PinIdea/ASIC-X11-Miner#8]
Issue: When restarting the miner using the web interface, the miner process will not always restart.

Steps to Reproduce:
* Operate the miner normally for some long period of time (1 week or so probably)
* Connect to the miner (possibly must be through a reverse proxy like Apache2)
* Navigate to http://example.com/config.html
* Change settings to mine from a different pool
* Click Save & Restart

Notes:
Observe the lack of any change to the miner's currently connected pool.

The behavior has been somewhat inconsistent here, so it's tough to say how easily this can be reproduced, but one change I would recommend is to always use the URI provided as the URI by the client in the HTTP server rather than using the DHCP/Static IP address when reconnecting.  These two things may or may not be related, and I can break it out into another issue if need be.

I'd try my hand at fixing the bug if any DR-3 code were on Github.

**Comment by papuchobello** [comment]
hi my pinidea dr-3 is restarting a lot and I dont detect it so I can not configure it.
I may ask you please for iso copy of the sd card inside the miner. I think my sd card sofware is corructed.
thank you
umanac@gmail.com

**Comment by fokilpt** [comment]
Ugh where is the SD Card slot at in the DR3?

**Comment by papuchobello** [comment]
the dr3 has the controller inside you have to open to get it. is really
easy is a micro sd card.
let me know if you need the sd controller image iso.  the default factory
setting.

good luck

**Comment by fokilpt** [comment]
Hey i could DEFINITELY use the default image iso! my email is fokilpt@gmail.com

is it under the green board that you can take off or?

**Comment by fokilpt** [comment]
Hey I can't find the controller? Like the sd card slot but yes I need the
iso asap look

It's not that little white chip is it? It's too skinny and wide to be a sd
card I put my micro SD next to it and it was longer and wider and fatter :/
I can't find it anywhere I don't have the iso anyway sadly

**Comment by papuchobello** [comment]
the controller is exactly the card were you plug your network cable.
if you look at your card completely to the right you have a hdmi video
conection next to the left you got your network conection next to the left
you got 2 usb conections  and one audio conection.
well next to the audio conection on the side of the card you got your micro
sdcard card slot with your micro sdcard inside push it gently to take the
micro sdcard out.
​
 DR3-4GB.zip
<https://drive.google.com/file/d/0BzelMZ7nzIZVNjVVQm5kQmZPS2M/view?usp=drive_web>
​

**Comment by papuchobello** [comment]
I used win32diskimager-1.0.0 to install to the miner

**Comment by papuchobello** [comment]
I used win32diskimager-1.0.0

**Comment by papuchobello** [comment]
did you get it?

**Comment by papuchobello** [comment]
you see that blue card with the big h3 on it?

**Comment by frobteam** [comment]
Yes, been responding to you via email if that was you? I tried flashing it using a program called "Etcher" but I am now goign to try win32diskimager-1.0.0 as you suggested.

When I tried flashing it before with what you linked, everything turned yellow (dont know if thats normal) neither of the lights came on and it did not detect it being connected to ethernet so something was wrong IDK what :/

Any clue? I am hoping the firmware or the win32diskimager will fix it this time.

**Thanks again @papuchobello for all your help!**

**Comment by fokilpt** [comment]
Yes, been responding to you via email if that was you? I tried flashing it using a program called "Etcher" but I am now goign to try win32diskimager-1.0.0 as you suggested.

When I tried flashing it before with what you linked, everything turned yellow (dont know if thats normal) neither of the lights came on and it did not detect it being connected to ethernet so something was wrong IDK what :/

Any clue? I am hoping the firmware or the win32diskimager will fix it this time.

**Thanks again @papuchobello for all your help!**

**Comment by fokilpt** [comment]
With win32diskimager-1.0.0 it won't even work, did you use a certain Hash? 

https://i.gyazo.com/ee1f58aee44b8408feaa95b52351aa9b.png

That is what I got using the win32diskimager with no settings changed. I have the ability to generate 3 different hashes but Dash is PoS... So none would be useful correct?

Going to try Etcher again at least it claimed to work. I went from being able to access the login page but not login to it not even detecting Ethernet now though :/ hope the firrmware helps.

Any suggestions?

**Comment by fokilpt** [comment]
I am trying to find my 32gb sd card right now so i can use it.... and i just thought of something, isnt there a way to format sd cards? going to look for that as well..

**Comment by papuchobello** [comment]
yes just insert your sd card in your pc or with an adaptor and format it (
fat32) just to be safe.

Try this

unzip the file that I have send to you.  it will have the .img at the end.
and try again with the unziped file.
I use 7-zip to unziped.

**Comment by fokilpt** [comment]
Yeah I didn't format it last time. Should I add the firmware?

**Comment by fokilpt** [comment]
Anyone? Firmware or no? still can not get this thing to work.

If anyone is interested I get my new Galaxy S7 Edge tomorrow and could walk you side by side what is going on live.

I will pay BTC for this.

Email me if so: fokilpt at gmail.com dot c()m :)
